packages/core/src/services/commits/runDocumentAtCommit.test.ts

```diff
@@ -0,0 +1,232 @@
+    })
+
+    expect(result.error).toBeDefined()
+  })
+
+  it('fails if documents has not valid config', async () => {
+    const { workspace, document, commit } = await buildData()
+    const result = await runDocumentAtCommit({
+      workspaceId: workspace.id,
+      document,
+      commit,
+      parameters: {},
+      providerLogHandler: mocks.providerLogHandler,
+    })
+
+    await expect(result?.value?.response).rejects.toThrowError(
+      'Error validating document configuration',
+    )
+  })
+
+  it('fails if provider api key is not found', async () => {
+    const { workspace, document, commit } = await buildData({
+      doc1Content: dummyDoc1Content,
+    })
+    const result = await runDocumentAtCommit({
+      workspaceId: workspace.id,
+      document,
+      commit,
+      parameters: {},
+      providerLogHandler: mocks.providerLogHandler,
+    })
+
+    await expect(result?.value?.response).rejects.toThrowError(
+      'ProviderApiKey not found',
+    )
+  })
+
+  describe('with an existing provider key', () => {
+    beforeEach(async () => {
+      const {
+        workspace: wsp,
+        user: usr,
+        document: doc,
+        commit: cmt,
+      } = await buildData({
+        doc1Content: dummyDoc1Content,
+      })
+      user = usr
+      document = doc
+      commit = cmt
+      workspaceId = wsp.id
+      workspace = wsp
+      provider = await factories.createProviderApiKey({
+        workspace,
+        type: Providers.OpenAI,
+        name: 'openai',
+        user,
+      })
+    })
+
+    it('returns document resolvedContent', async () => {
+      const result = await runDocumentAtCommit({
+        workspaceId,
+        document,
+        commit,
+        parameters: {},
+        providerLogHandler: mocks.providerLogHandler,
+        // @ts-ignore
+        runAi: mocks.runAi,
+      })
+
+      expect(result.value?.resolvedContent.trim()).toEqual(`---
+provider: openai
+model: gpt-4o
+---
+
+
+This is a test document
+<step />`)
+    })
+
+    it('returns document log UUID', async () => {
+      const result = await runDocumentAtCommit({
+        workspaceId,
+        document,
+        commit,
+        parameters: {},
+        providerLogHandler: mocks.providerLogHandler,
+        // @ts-ignore
+        runAi: mocks.runAi,
+        // @ts-ignore
+        generateUUID: mocks.uuid,
+      })
+
+      expect(result.value?.documentLogUuid).toEqual('fake-document-log-uuid')
+    })
+
+    it('pass documentUuid to AI', async () => {
+      const { runDocumentAtCommit } = await import('./index')
+      const { stream } = await runDocumentAtCommit({
+        workspaceId,
+        document,
+        commit,
+        parameters: {},
+        // @ts-ignore
+        providerLogHandler: mocks.providerLogHandler,
+        // @ts-ignore
+        generateUUID: mocks.uuid,
+      }).then((r) => r.unwrap())
+
+      await consumeStream(stream)
+      expect(mocks.runAi).toHaveBeenCalledWith(
```

Comment:
Mocks are hard but on top of that I was not waiting for the stream to be consumed so the test was not passing 🤯